lms/djangoapps/commerce/views.py

```diff
@@ -148,9 +149,37 @@ def checkout_cancel(_request):
 @login_required
 def checkout_receipt(request):
     """ Receipt view. """
+
+    page_title = _('Receipt')
+    is_payment_complete = True
+    is_cybersource = all(k in request.POST for k in ('signed_field_names', 'decision', 'reason_code'))
+    if is_cybersource and request.POST['decision'] != 'ACCEPT':
+        # Cybersource may redirect users to this view if it couldn't recover
+        # from an error while capturing payment info.
+        is_payment_complete = False
+        page_title = _('Payment Failed')
+        reason_code = request.POST['reason_code']
+        # if the problem was with the info submitted by the user, we present more detailed messages.
+        is_user_payment_error = (200 <= int(reason_code) <= 233) or int(reason_code) in (101, 102, 240)
+        if is_user_payment_error:
+            error_summary = _("There was a problem with the payment information you submitted.")
+            error_text = REASONCODE_MAP[request.POST['reason_code']]
```

Comment:
REASONCODE_MAP is actually a [defaultdict](https://docs.python.org/2/library/collections.html#collections.defaultdict).  I considered changing that but I'd prefer to keep the scope of this PR as tight as I can.  Since it's a defaultdict we will need to check the key and value of the mapping to be sure we have usable message.  (some changes related to this in my next update)